code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -844,6 +844,96 @@
 
 	return ..()
 
+/obj/effect/alien/resin/destroyer_cocoon
+	name = "alien cocoon"
+	desc = "A large pulsating cocoon."
+	health = 3600
+	icon = 'icons/obj/structures/alien/xenoDestroyerHatchery.dmi'
+	icon_state = "static"
+
+	var/timer
+	var/mob/chosen_candidate
+
+/obj/effect/alien/resin/destroyer_cocoon/Initialize(mapload, pylon)
+	. = ..()
+	timer = addtimer(CALLBACK(src, PROC_REF(start_growing)), 10 SECONDS, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME)
+	addtimer(CALLBACK(src, PROC_REF(check_pylons)), 10 SECONDS, TIMER_UNIQUE)
+
+/obj/effect/alien/resin/destroyer_cocoon/proc/check_pylons()
+	var/datum/hive_status/hive = GLOB.hive_datum[XENO_HIVE_NORMAL]
+	
+	if(hive.active_endgame_pylons.len < 2)
+		qdel(src)
+	
+	addtimer(CALLBACK(src, PROC_REF(check_pylons)), 10 SECONDS, TIMER_UNIQUE)
+
+
+/obj/effect/alien/resin/destroyer_cocoon/proc/start_growing()
+	icon_state = "growing"
+	timer = addtimer(CALLBACK(src, PROC_REF(choose_candidate)), 9 MINUTES, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME)
+
+/obj/effect/alien/resin/destroyer_cocoon/proc/roll_candidates()
+	// First, Let the Queen choose
+	var/datum/hive_status/hive = GLOB.hive_datum[XENO_HIVE_NORMAL]
+	var/mob/living/carbon/xenomorph/choice = tgui_input_list(hive.living_xeno_queen, "Choose a xenomorph to become the destroyer", "Choose a xeno", hive.totalXenos - hive.living_xeno_queen)
+
+	if(choice)
+		return choice
+
+	// Then ask all the living xenos
+	var/list/total_xenos_copy = shuffle(hive.totalXenos.Copy())
+
+	for(var/mob/living/carbon/xenomorph/candidate in total_xenos_copy)
+		if (get_job_playtime(candidate.client, JOB_XENO_ROLES) < 25 HOURS)
```

Comment:
would having a timelock based on T3 hours be better?

Reply:
Don't think so
New players play and suicide with T3s plenty